app/components/channel_link/channel_link.js

```diff
@@ -55,7 +89,12 @@ export default class ChannelLink extends React.PureComponent {
             return <Text style={this.props.textStyle}>{`~${this.props.channelName}`}</Text>;
         }
 
-        const suffix = this.props.channelName.substring(channel.name.length);
+        let suffix;
+        if (!channel.name && !channel.id && channel.display_name) {
+            suffix = this.props.channelName.substring(channel.display_name.length);
```

Comment:
This should also use the name matched by `getChannelFromChannelName` or else it won't correctly set `suffix` to be any punctuation trailing after the channel name

Reply:
Okay, so what I did is just same rendering as the original. In another case where a user got channel mention for channel that he's not currently member of, just render the display name.

![screen shot 2019-02-06 at 6 02 50 pm](https://user-images.githubusercontent.com/5334504/52334322-5d1a8680-2a3a-11e9-8990-2a46fc478f31.png)
In above screenshot, 'test2' user is not member of channel 'one' and 'three'. Channel 'one' is {name: 'one', display_name: 'one'} while channel three is {name: 'three-three', display_name: 'three'}.

Reply:
The problem with this is that it won't work if the channel name and display name are different lengths. If you have a channel like `{name: 'name', display_name: 'Display Name'}` and you post `~name.`, it will cut off the period. Similarly, if you have `{name: 'longname', display_name: 'Short'}` and post `~longname.`, it'll render like `~Shortame.`